create Models folder, containing json file with restaurant data

jamilazd opened the pull request:
# Pull Request Template

## Created temporary JSON of restaurants 

Breakdown of Task, steps taken:

1.  I looked at examples of JSON data, such as the JSON file used in our bookcase app project. 

2.  I created a JSON file in VS code and named it restaurants.json.  I then created a folder called Models and moved the 
     JSON file into it. 

3. I created a structure for the restaurants.json file, using the bookcase app JSON as a guide. I included restaurant name, 
    address, phone number, Trip Advisor rating, website, email, Facebook, Instagram, cuisine type and image links. I referred to 
    the information listed on the Restaurant Review Spreadsheet to populate the restaurants.json file. 

5. I checked that the restaurants.json contained valid JSON, using an online validator (https://jsonlint.com/) 

6.  I created forked the group project repo and cloned it locally. I then moved my models folder, containing the json file into 
     the SRC folder. I committed the changes locally and then pushed these changes. 


:question: To complete the changes, did you use any tutorials or do any reading that would be of benefit to the group? It is great for our growth, and can also aid with troubleshooting.

-  http://oss-watch.ac.uk/resources/pullrequest

- https://jsonlint.com/

- https://www.w3schools.com/js/js_json_syntax.asp

## Issue Addressed

:question: This pull request is in response to **issue #19**

:question: Does it require the install of any other dependencies apart from the current node_modules when a **npm install** is run? If yes, please list here. **No**

## Type of change

:question: Please tick all relevant options

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x]  **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **This change requires a documentation update**

## Has This Been Tested?

I checked the validity of my data, using an online JSON validator: https://jsonlint.com/  and it came back as valid JSON


## Checklist:

:question: For peace of mind.

- [x] My code follows the guidelines of this project
- [x] I have used appropriate and relevant pull request title
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas  **N/A**
- [x] I have made corresponding changes to the documentation or I have not needed to
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [ ] Linked issues  **N/A**
- [x] Assigned team or specific members to review.

jamilazd commented:
Hi! I can see there is an error above regarding my pull request. I'm not sure whether it's because I selected reviewers after I submitted the request.

tanyapowell commented:
It's because you forked the repository @jamilazd ([see this for more details](https://github.com/actions/first-interaction/issues/10)) instead of cloning it, and just means you won't get a greeting on your first PR. Don't worry about the greetings action failing, it would've just been a message saying thank you for making your first PR in the project 😄

jamilazd commented:
Oops sorry, I had no idea I'd done that! I'll read up on the differences between forking and cloning, so I can understand.

emma-mendez commented:
Hows it going Jam? Wish I was in a position to offer sound help, but Im sure here for the support, Go Girl !!! You need anything let us know -

hawa-bah commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+            "landline": null, 
+            "mobile": "07792557204"
+        },
+        "cusineCategory": "Trinidadian, dine-in and takeaway", 
```
Maybe cusineCategory could be arranged in a different way. It seems like a description at the moment, and it could be helpful to have it in an array. Apart from that, I love the structure of the Json file! Great work!

jamilazd replied:
Thanks Hawa! That's a great idea, it makes more sense to arrange it in an array like you said, I'll get on and change it. The array could include things like whether you can dine in or not, whether they offer takeaway and type of cuisine.

hawa-bah replied:
sounds great!

emma-mendez commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+            "landline": null, 
+            "mobile": "07792557204"
+        },
+        "cusineCategory": "Trinidadian, dine-in and takeaway", 
```
Hi J, these are brilliant so far, I'm wondering if we could separate the categories so to help with filtering later. so dine in or take away would be separate to origin/ country of cuisine. If searching for take away Jamaican or dine in Trinidadian or just African as a category.

jamilazd replied:
Thanks, that's a good idea! Hawa suggested that I makes this into separate arrays.

emma-mendez commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+        "cusineCategory": "Trinidadian, dine-in and takeaway", 
+        "tripAdvisorRating": 5, 
+        "website": null, 
+        "instgram": null,
```
dont forget the a in instagram

jamilazd replied:
Oops I've gone back and corrected this

hawa-bah commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+        "cusineCategory": "Caribbean dine-in", 
+        "tripAdvisorRating": 4.5, 
+        "website": "https://www.devonhouse.co.uk/", 
+        "instgram": "devonhousejq",
```
maybe a part from the user name, we could also add the full URL path of instagram, and facebook. It will be helpful when creating a link to an icon

AffirmedVisionary replied:
we could leave it as username and in react create a function that would append the username in the database to the end of the URL. would save repetition.

tanyapowell replied:
Good thinking @AffirmedVisionary! The first part of the url will always be `https://instagram.com/` so that can hardcoded in, and then we just append the username at the end like you said, e.g. `https://instagram.com/${instagram}`. Wouldn't even need to store it in the database because it _should_ just be needed on the frontend to create the link

emma-mendez commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+            "landline": null, 
+            "mobile": "07792557204"
+        },
+        "cusineCategory": "Trinidadian, dine-in and takeaway", 
```
This may be more of a description?

hawa-bah reviewed (commented):
Hi Jamila, your work looks great! I've added some comments .

AffirmedVisionary commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+            "mobile": null
+        },
+        "cusineCategory": "Plant based Caribbean, dine-in and takeaway", 
+        "tripAdvisorRating": 5, 
```
not really a comment but was wondering if anyone knows if trip advisor has an API where we could create a get request where it searches for a name and location match then automatically returns the rating. Users leaving our app to go look at trip advisor maybe an issue.

jamilazd replied:
I had a look and TripAdvisor have an API for developers called the Content API. 
http://developer-tripadvisor.com/content-api/ 

They also have widgets:  https://www.tripadvisor.com/Widgets

AffirmedVisionary replied:
ooh that is good to know! thanks for looking.

tanyapowell commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+[  
+    {
+        "restaurantName": "MBM Trinbago Meals",
+        "address": "Pool Road, Birmingham B66 3DB",
```
Can I be really annoying and ask for the address to be an object? Will help eventually when it needs to be stored in the database, for any location stuff you wanna do in the future and even displaying the address on the frontend.
 E.g. 
```
"address": {
  "firstLine": "Pool Road",
  "secondLine": null,
  "city": "Birmingham",
  "postcode": "B66 3DB"
}
```

tanyapowell commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+        "tripAdvisorRating": 5, 
+        "website": null, 
+        "instgram": null,
+        "facebook": "@mbmmeals", 
```
Probably worth removing the `@`

tanyapowell commented on src/Models/restaurants.json:
```diff
@@ -0,0 +1,141 @@
+
+    }
+
+]
```
Need to add an empty line at the end of this file 😄

EbonyChantelle reviewed (commented):
Hey @jamilazd! This looks great. Happy to approve once the minor changes have been made i.e. address as an object, removal of the @ and space added at the end.

In addition, in terms of the cuisine category, ditto what @iesea and @hawa-bah said about the separating it to allow for more specific filtering on search’s i.e. cuisineCategory: Caribbean and then Options (though maybe need a better word): dine in, take away, delivery etc etc or Options: dine in: yes, take away: no, delivery: no.  If that makes sense?

Also, keep spelling and capital casing consistent i.e. cuisineCategory rather than cusineCategory, and Plant Based Caribbean rather than Plant based Caribbean.

jamilazd commented:
As recommended, I've removed the @ from the Instagram and Facebook handles and removed the first part of the URL for the instagram and facebook image links. I've also changed the address and cuisine type sections into objects. I've read through and double checked/ corrected spelling. Hopefully there are no more spelling errors (I'm a terrible speller) Please let me know if there are any more changes you'd like me to make

EbonyChantelle reviewed (approved):
I think everything is sorted here from what I can see.
